nixos/modules/services/hardware/irexec.nix

```diff
@@ -0,0 +1,119 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
```

Comment:
Can you try to prevent such a wide-scoped with-statement?
You can import the parts of lib, that you need with something like:

```nix
let
   inherit (lib) mkEnableOption ... ;
in
```

Reply:
I have no idea how to do it right, but in [modules/services/hardware/lirc.nix#L3](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/hardware/lirc.nix#L3) they do exactly the same:

    { config, lib, pkgs, ... }:

    with lib;

    let

Reply:
There are a lot of old (and sometimes even new) packages and modules that do this, but there is an ongoing effort to remove filewide `with lib;` statements. Have a quick look at the output of `git log --grep "with lib"`.